### PR TITLE
salary calculation task

### PR DESCRIPTION
**Pull request by ankit-tag97**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -8,35 +8,36 @@ datasource db {
...
+  pincode              String               @db.VarChar
+  profileImageUrl      String               @map("profile_image_url") @db.VarChar
+  aadhar               String               @db.VarChar
+  otherIdProof         String               @map("other_id_proof") @db.VarChar
+  isActive             Boolean?             @default(true)
   password             String?
   employeeAsset        EmployeeAsset[]
   employeeRole         EmployeeRole[]
-  EmployeeInvoce       EmployeeInvoice[]
+  EmployeeInvoce       EmployeeSalarySlip[]
```
Comment (dhruvitmakadia) [inline prisma/schema.prisma]:
```suggestion
  EmployeeInvoice       EmployeeSalarySlip[]
```

File: prisma/schema.prisma [hunk 2]
```diff
@@ -111,23 +112,35 @@ model EmployeeInvoice {
...
-  paymentTerms      Int       @default(0) @map("payment_term")
-  totalAmountPaid   Float     @default(0) @map("total_amount_paid")
-  paymentDeu        DateTime  @db.Timestamptz(6)
+  netPay            Float?    @default(0.00) @map("net_pay")
 
-  @@map("employees_invoce")
+  @@map("employee_salary_slip")
+}
+
+model salary {
```
Comment (dhruvitmakadia) [inline prisma/schema.prisma]:
```suggestion
model Salary {
```

File: src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { Body, Controller, Delete, Get, NotFoundException, Param, Patch, Post, UseFilters } from '@nestjs/common';
+import { EmployeeSalaryDto } from '../../dtos/employee-salary.dto';
+import { EmployeeSalaryService } from '../../service/employee-invoice/employee-salary.service';
+import { DataNotFoundException } from '../../../exception/DataNotFoundException';
+import { IdExceptionFilter } from '../../../exception/id-exception-filter';
+
+@Controller('employee-salary')
```
Comment (dhruvitmakadia) [inline src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts]:
```suggestion
@Controller('salary')
```

File: src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+// @UseFilters(IdExceptionFilter)
+export class EmployeeSalaryController {
+    constructor(private readonly salaryService: EmployeeSalaryService) { }
```
Comment (dhruvitmakadia) [inline src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts]:
```suggestion
    constructor(private readonly salaryService: SalaryService) { }
```

File: src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Post('/create')
```
Comment (dhruvitmakadia) [inline src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts]: we can say generate

File: src/employee-invoice/dtos/employee-salary.dto.ts [hunk 6]
```diff
@@ -1,13 +1,13 @@
-import { IsDateString, IsEmail, IsNotEmpty, IsNumber, IsString } from "class-validator";
-import { Base } from "src/interface/base";
+import { IsDateString, IsEmail, IsNotEmpty, IsNumber, IsOptional, IsString } from "class-validator";
+import { Base } from "../../interface/base";
+
+export class EmployeeSalaryDto extends Base {
```
Comment (dhruvitmakadia) [inline src/employee-invoice/dtos/employee-salary.dto.ts:4]: FE has only this 5 things

Emp id, Overtime hour, Absent Hour, advanceWithdrawal and Incentive

File: src/employee-invoice/service/employee-invoice/employee-salary.service.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+@Injectable()
+export class EmployeeSalaryService {
+    constructor(private readonly prisma: PrismaService) { }
+
+    async createInvoice(data: Prisma.EmployeeSalarySlipCreateManyInput) {
+        const totalAmount = data.basicSalary + data.insentive +
+            (data.overTime * ((data.basicSalary / 30) / 9)) -
+            (data.absent * (data.basicSalary / 30) / 9) -
+            data.professionalTax - (data.securityAmount * (1 * 0.1)) - data.paidLeave - data.advanceWithdrawal
```
Comment (dhruvitmakadia) [inline src/employee-invoice/service/employee-invoice/employee-salary.service.ts]:
```suggestion
            data.professionalTax - (data.securityAmount * 0.1) - data.paidLeave - data.advanceWithdrawal
```

File: src/employee-invoice/service/employee-invoice/employee-salary.service.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import { EmployeeSalaryDto } from '../../dtos/employee-salary.dto';
+import { PrismaService } from '../../../prisma/prisma.service';
+
+@Injectable()
+export class EmployeeSalaryService {
+    constructor(private readonly prisma: PrismaService) { }
+
+    async createInvoice(data: Prisma.EmployeeSalarySlipCreateManyInput) {
+        const totalAmount = data.basicSalary + data.insentive +
+            (data.overTime * ((data.basicSalary / 30) / 9)) -
```
Comment (dhruvitmakadia) [inline src/employee-invoice/service/employee-invoice/employee-salary.service.ts]: Store this in seperate varible and reuse all place
data.basicSalary / 30) / 9

like

const hourlyRate = data.basicSalary / 30) / 9;

Note: Calculation based on selected month wise For example we are calculating for August-2023 so after remove all sat-sun we have 23 working days so hourly rate calculated on that

**Review by dhruvitmakadia — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma (hunk 1 above)
Comment (ankit-tag97) [reply prisma/schema.prisma]: done

File: prisma/schema.prisma (hunk 2 above)
Comment (ankit-tag97) [reply prisma/schema.prisma]: done

File: src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts (hunk 5 above)
Comment (ankit-tag97) [reply src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts]: done

File: src/employee-invoice/service/employee-invoice/employee-salary.service.ts (hunk 7 above)
Comment (ankit-tag97) [reply src/employee-invoice/service/employee-invoice/employee-salary.service.ts]: DONE

File: src/employee-invoice/service/employee-invoice/employee-salary.service.ts (hunk 8 above)
Comment (ankit-tag97) [reply src/employee-invoice/service/employee-invoice/employee-salary.service.ts]: done

File: src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts (hunk 3 above)
Comment (ankit-tag97) [reply src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts]: done

File: src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts (hunk 4 above)
Comment (ankit-tag97) [reply src/employee-invoice/controller/employee-invoice/employee-salary.controller.ts]: done

**Pull request merged**